src/workerd/api/actor-state.c++

```diff
@@ -752,6 +753,30 @@ jsg::Promise<jsg::Value> DurableObjectState::blockConcurrencyWhile(jsg::Lock& js
   return IoContext::current().blockConcurrencyWhile(js, kj::mv(callback));
 }
 
+void DurableObjectState::acceptWebSocket(jsg::Ref<WebSocket> ws, kj::Array<kj::String> tags) {
+  JSG_ASSERT(!ws->isAccepted(), Error,
+      "Cannot call `acceptWebSocket()` if the WebSocket was already accepted via `accept()`");
+  JSG_ASSERT(ws->LOCAL == WebSocket::Locality::LOCAL, Error,
+      "Outgoing WebSocket connections cannot be hibernated.");
+
+  // We need to get a HibernationManager to give the websocket to.
+  auto& a = KJ_REQUIRE_NONNULL(IoContext::current().getActor());
+  if (a.getHibernationManager() == nullptr) {
+    // TODO(now): Actually set the hibernation manager.
```

Comment:
FYI this `TODO(now)` is replaced in a later commit once we actually have a `HibernationManagerImpl`.